distributed/shuffle/_scheduler_extension.py

```diff
@@ -161,27 +171,22 @@ def _create_dataframe_shuffle_state(
         schema = spec["schema"]
         column = spec["column"]
         npartitions = spec["npartitions"]
+        parts_out = spec["parts_out"]
         assert schema is not None
         assert column is not None
         assert npartitions is not None
+        assert parts_out is not None
 
         workers = list(self.scheduler.workers)
         output_workers = set()
 
-        name = barrier_key(id)
         mapping = {}
 
-        for ts in self.scheduler.tasks[name].dependents:
-            part = get_partition_id(ts)
-            if ts.worker_restrictions:
-                output_worker = list(ts.worker_restrictions)[0]
-            else:
-                output_worker = get_worker_for_range_sharding(
-                    part, workers, npartitions
-                )
+        for part in parts_out:
+            # TODO: How do we deal with pre-existing worker restrictions?
```

Comment:
why can't this logic be the same as before when it comes to existing worker restrictions?

Reply:
I've adjusted the logic. The only unhandled situation I can imagine is if a task is fused with a follow-up task, which has worker restrictions of its own. In this case, we may ignore those. A fix for this behavior would be adding a shuffle-internal mechanism for transferring the output partition to another worker.